packages/waku/src/router/client.ts

```diff
@@ -33,7 +33,9 @@ import {
 import type { RouteProps, ShouldSkip } from './common.js';
 import type { RouteConfig } from './base-types.js';
 
-type InferredPaths = RouteConfig extends { paths: infer UserPaths }
+type InferredPaths = RouteConfig extends {
+  paths: { expanded: infer UserPaths };
```

Comment:
I don't particularly like this change. `expanded` is anything meaningful here.
As I understand, a change here is necessary to simplify the type overload.
If we can't avoid it, how about this?
 ```suggestion
   paths: { inferred: infer UserPaths };
```

Reply:
so there is also this

```ts
export type PagePath<Config> = Config extends {
  paths: {
    initial: infer AllPages;
  };
}
  ? AllPages
  : never;
```

the key thing is that I need to differentiate the inferred routes that are with the slugs and the ones with the slugs replaced with staticPaths or `string`. This is why I went with expanded and initial: 

```ts
export type PathsForPages<PagesResult extends { DO_NOT_USE_pages: AnyPage }> = {
  /** All possible paths with string or static paths in place of slug */
  expanded: CollectPaths<PagesResult['DO_NOT_USE_pages']> extends never
    ? string
    : CollectPaths<PagesResult['DO_NOT_USE_pages']> & {};
  /** Paths with slugs as string literals */
  initial: PagesResult['DO_NOT_USE_pages']['path'];
};
```

I am happy to rename these, but I fear `inferred` is even more vague than the current names since they are both technically inferred types from the base of `RouteConfig`

Reply:
`PagePath` is for `createPages`, while `InferredPaths` is for `defineRouter` which isn't tied to `ceatePages`.

`PathsForPages` is creating for both and my concern is just naming, because `defineRouter` is designed to be used with something else other than `createPages`, currently.

Reply:
Yeah, I share that concern... I'll think on new names 🤔, but please feel free to suggest other ideas if you have them :)

Reply:
> to simplify the type overload.

Maybe, I prefer separating overloads. (Your original idea.)

```ts
    interface RouteConfig {
      paths: PathsForPages<typeof _pages>;
      something: SomethingForPages<typeof _pages>;
    }
```

They are for different purposes and consumed differently. So, explicit separation isn't that bad. Besides, this code will usually auto generated.

Reply:
Or, maybe this is more correct:
```ts
    interface RouteConfig {
      paths: PathsForPages<typeof _pages>;
    }
    interface PagesConfig {
      pages: typeof _pages;
    }
```

Reply:
I like this second example as well. Having two interfaces is a downside, but having the flexibility to define other types of router definitions like defineEntries later with this pattern shows the benefit to me.